bba/bba.go

```diff
@@ -3,92 +3,131 @@ package bba
 import (
 	"encoding/json"
 	"errors"
+	"strconv"
 	"sync"
 	"sync/atomic"
 
-	"github.com/DE-labtory/cleisthenes/pb"
 	"github.com/golang/protobuf/ptypes"
 
-	"github.com/DE-labtory/cleisthenes"
-)
-
-var ErrUndefinedRequestType = errors.New("unexpected request type")
-
-type Binary = bool
+	"github.com/DE-labtory/cleisthenes/pb"
 
-const (
-	one  Binary = true
-	zero        = false
+	"github.com/DE-labtory/cleisthenes"
 )
 
 type request struct {
 	sender cleisthenes.Member
-	data   *pb.Message_Bba
+	data   cleisthenes.Request
+	round  uint64
 	err    chan error
 }
 
 type BBA struct {
 	*sync.RWMutex
+	cleisthenes.Tracer
 
+	owner cleisthenes.Member
 	// number of network nodes
 	n int
-
 	// number of byzantine nodes which can tolerate
 	f int
 
-	proposer cleisthenes.Member
-
 	stopFlag int32
 	// done is flag whether BBA is terminated or not
 	done bool
```

Comment:
It can occur data race. `done` also can be `BinaryState`

Reply:
good! 👍 and fixed!